src/Caching/Impl/ModuleDatabase.cs

```diff
@@ -191,7 +208,52 @@ public Task StoreModuleAnalysisAsync(IDocumentAnalysis analysis, CancellationTok
             return _fs.FileExists(dbPath) ? dbPath : null;
         }
 
+        private bool FindModuleModel(string moduleName, string filePath, out ModuleModel model) {
+            model = null;
+
+            // We don't cache results here. Module resolution service decides when to call in here
+            // and it is responsible of overall management of the loaded Python modules.
+            for (var retries = 50; retries > 0; --retries) {
+                try {
+                    // TODO: make combined db rather than per module?
+                    var dbPath = FindDatabaseFile(moduleName, filePath);
+                    if (string.IsNullOrEmpty(dbPath)) {
+                        return false;
+                    }
+
+                    using (var db = new LiteDatabase(dbPath)) {
+                        if (!db.CollectionExists("modules")) {
```

Comment:
Are there async versions of these methods?

Reply:
No. Should we have them?

Reply:
If LiteDB doesn't provide async IO, then there is not much we can do.